modules/gapi/src/backends/fluid/gfluidbackend.cpp

```diff
@@ -984,9 +990,10 @@ namespace
 
                 auto &fu = fg.metadata(node).get<FluidUnit>();
 
-                GAPI_Assert((out_ws.size() == 1 && out_hs.size() == 1) &&
+                GAPI_Assert(((out_ws.size() == 1 && out_hs.size() == 1) &&
```

Comment:
What does this assert stand for? I'm OK with a single output mat, but why it checks for exactly 1(2) input mats?

Reply:
As discussed with Dima Matveev, there is no need to use fluid without mats (for a single row processing?). So, need to put static assert somewhere(probably macros) before this one, to check if we have at least 1 input mat in kernel.

Reply:
Also you get this macro wrong.

It doesn't check how many input/output objects there, it checks if _dimensions_ of inputs/outputs are equal.

GArray doesn't (shouldn't) contribute here in any form.

Reply:
Got it, thanks. Removed the logic